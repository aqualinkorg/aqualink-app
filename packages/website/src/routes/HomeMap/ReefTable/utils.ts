import { startCase } from "lodash";
import type { TableRow } from "../../../store/Homepage/types";

export type Order = "asc" | "desc";

export enum OrderKeys {
  ALERT = "alert",
  DEPTH = "depth",
  DHW = "dhw",
  LOCATION_NAME = "locationName",
  TEMP = "temp",
}

// This type isn't used anywhere, it just forces the above enum to only hold valid keys to TableRow.
type __CheckOrderKeys = TableRow[OrderKeys];

/**
 * Returns a human friendly string of a order key.
 * @param orderKey key to return the string for.
 */
export function getOrderKeysFriendlyString(orderKey: OrderKeys) {
  switch (orderKey) {
    case OrderKeys.DHW:
      return "DHW";
    default:
      return startCase(orderKey);
  }
}

export function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
  return b[orderBy] >= a[orderBy] ? 1 : -1;
}

export function getComparator(
  order: Order,
  orderBy: OrderKeys
): (
  a: {
    [key in OrderKeys]: number | string | null;
  },
  b: {
    [key in OrderKeys]: number | string | null;
  }
) => number {
  return order === "desc"
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

export function stableSort<T>(array: T[], comparator: (a: T, b: T) => number) {
  // eslint-disable-next-line fp/no-mutating-methods
  return [...array].sort(comparator);
}
