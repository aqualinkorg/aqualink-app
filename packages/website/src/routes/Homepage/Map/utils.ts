export const coloredBuoy = (color: string) =>
  `<?xml version="1.0" encoding="UTF-8"?>
  <svg width="29px" height="35px" viewBox="0 0 29 35" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
      <defs>
          <path d="M10.0023993,-6.25277607e-13 C17.4094117,-6.25277607e-13 22.207967,7.82436546 18.9739831,14.3966715 C17.4094871,17.7347423 12.1924976,22.7424773 10.0023993,25.1417547 C7.70723368,22.5332976 2.80456638,17.9439217 1.13490246,14.5007584 C-2.30725517,8.03253919 2.49117441,-6.25277607e-13 10.0023993,-6.25277607e-13 Z" id="path-1"></path>
          <filter x="-35.0%" y="-19.9%" width="170.0%" height="155.7%" filterUnits="objectBoundingBox" id="filter-2">
              <feOffset dx="0" dy="2" in="SourceAlpha" result="shadowOffsetOuter1"></feOffset>
              <feGaussianBlur stdDeviation="2" in="shadowOffsetOuter1" result="shadowBlurOuter1"></feGaussianBlur>
              <feColorMatrix values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.5 0" type="matrix" in="shadowBlurOuter1"></feColorMatrix>
          </filter>
      </defs>
      <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
          <g id="Reef-Detail-full---no-live-feed-Copy-2" transform="translate(-748.000000, -491.000000)">
              <g id="icon_sat" transform="translate(752.500000, 493.500000)">
                  <g id="Fill-1">
                      <use fill="black" fill-opacity="1" filter="url(#filter-2)" xlink:href="#path-1"></use>
                      <use fill="${color}" fill-rule="evenodd" xlink:href="#path-1"></use>
                  </g>
                  <path d="M15.2680598,9.87791599 C11.5105654,13.6354103 6.18310234,8.31883808 9.9461188,4.555975 L12.1649364,6.77540616 L12.1649364,6.41409093 C12.8785627,6.41409093 13.4591827,6.99710385 13.4591827,7.7107302 L13.0996622,7.70953221 L15.2680598,9.87791599 Z M16.5730897,7.82384158 C16.5736894,5.33842841 14.5375756,3.3 12.0515489,3.3 L12.0515489,4.19818779 C14.0402783,4.19818779 15.6772794,5.83150751 15.6772794,7.82085043 L16.5730897,7.82324642 L16.5730897,7.82384158 Z M15.0787125,7.78729129 C15.0787125,6.1436943 13.7305335,4.79553061 12.0869518,4.79553061 L12.0869518,5.69431663 C13.2361966,5.69431663 14.1799878,6.63803121 14.1799878,7.78735265 L15.0787125,7.78729129 Z M14.2254993,14.820015 L13.1571229,15.8883607 C12.6082704,16.4372131 11.7100826,16.4372131 11.1611841,15.8883607 L9.19592376,13.9231004 L11.2625763,11.8564478 L14.2254993,14.8193708 L14.2254993,14.820015 Z M4.90689491,5.50141065 L3.8385495,6.56978704 C3.28969701,7.11863953 3.28969701,8.01682732 3.8385495,8.56572583 L5.80380954,10.5309862 L7.87046211,8.46433361 L4.90753915,5.50141065 L4.90689491,5.50141065 Z M7.7537,13.1817868 L6.98793566,13.9475511 C6.76803266,14.1668559 6.40912568,14.1668559 6.18922268,13.9475511 L5.77877594,13.5371044 C5.55947117,13.3172014 5.55947117,12.9582944 5.77877594,12.7383914 L6.54454028,11.9726271 C6.58228856,12.0427319 6.63082337,12.1086429 6.68954343,12.1673584 L7.55837196,13.0361869 C7.61709201,13.094907 7.68300309,13.1434418 7.75250507,13.1811901 L7.7537,13.1817868 Z M10.5345618,11.2584025 L9.13185186,12.6610971 C8.80230414,12.9906448 8.26362156,12.9906448 7.93407384,12.6610971 L7.06524531,11.7922685 C6.73569759,11.4627208 6.73569759,10.9240382 7.06524531,10.5944905 L8.50748441,9.15225143 C8.86340024,10.0426469 9.65732738,10.8653351 10.5345618,11.2578656 L10.5345618,11.2584025 Z" id="Fill-1" fill="#FFFFFF" transform="translate(10.000000, 9.800000) scale(-1, -1) translate(-10.000000, -9.800000) "></path>
              </g>
          </g>
      </g>
  </svg>`;
