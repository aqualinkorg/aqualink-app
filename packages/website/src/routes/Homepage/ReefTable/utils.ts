export type Order = "asc" | "desc";

export type OrderKeys = "locationName" | "temp" | "depth" | "dhw" | "alert";

export function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
  return b[orderBy] >= a[orderBy] ? 1 : -1;
}

export function getComparator(
  order: Order,
  orderBy: OrderKeys
): (
  a: {
    [key in OrderKeys]: number | string | null;
  },
  b: {
    [key in OrderKeys]: number | string | null;
  }
) => number {
  return order === "desc"
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

export function stableSort<T>(array: T[], comparator: (a: T, b: T) => number) {
  return [...array].sort(comparator);
}
