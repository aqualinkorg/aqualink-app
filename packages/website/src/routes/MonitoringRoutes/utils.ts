import { OptionsObject, SnackbarKey, SnackbarMessage } from 'notistack';
import React from 'react';
import { User } from 'store/User/types';

interface FetchDataProps<T> {
  user: User | null;
  enqueueSnackbar: (
    message: SnackbarMessage,
    options?: OptionsObject | undefined,
  ) => SnackbarKey;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
  setResult?: React.Dispatch<React.SetStateAction<T | null>>;
  getResult: (token: string) => Promise<T>;
}

export async function fetchData<T>({
  user,
  enqueueSnackbar,
  setLoading,
  setResult,
  getResult,
}: FetchDataProps<T>) {
  const { token } = user || {};
  if (!token) {
    enqueueSnackbar('User is not authenticated!', {
      variant: 'error',
    });
    return;
  }

  if (setResult) {
    setResult(null);
  }

  setLoading(true);
  try {
    const data = await getResult(token);

    if (setResult) {
      setResult(data);
    }
  } catch (error: any) {
    const message = error?.response?.data?.message || error?.message;
    const errorMessage =
      typeof message === 'object' ? message?.join('. ') : message;
    enqueueSnackbar(errorMessage || 'Request failed', {
      variant: 'error',
    });
  } finally {
    setLoading(false);
  }
}

export const includes = (
  a: string | undefined | null,
  b: string | undefined,
) => {
  if (!b?.trim()) return true;
  if (!a) return false;

  return a.toLocaleLowerCase().includes(b.toLocaleLowerCase());
};
