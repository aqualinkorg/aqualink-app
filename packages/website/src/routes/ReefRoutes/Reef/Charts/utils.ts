import type { Data } from "../../../../store/Reefs/types";

export const createDatasets = (dailyData: Data[]) => {
  const dailyDataLen = dailyData.length;

  const bottomTemperature = dailyData.map((item) => item.avgBottomTemperature);
  const surfaceTemperature = dailyData
    .filter(
      (item) =>
        item.surfaceTemperature !== null || item.satelliteTemperature !== null
    )
    .map((item) => item.surfaceTemperature || item.satelliteTemperature);

  const meanBottomTemperature =
    bottomTemperature.reduce((a, b) => a + b) / dailyDataLen;
  const meanSurfaceTemperature =
    surfaceTemperature.reduce((a, b) => a + b) / dailyDataLen;

  return {
    bottomTemperatureData: [
      meanBottomTemperature,
      ...bottomTemperature,
      meanBottomTemperature,
    ],
    surfaceTemperatureData: [
      meanSurfaceTemperature,
      ...surfaceTemperature,
      meanSurfaceTemperature,
    ],
  };
};

export const calculateAxisLimits = (dailyData: Data[]) => {
  const dates = dailyData
    .filter(
      (item) =>
        item.surfaceTemperature !== null || item.satelliteTemperature !== null
    )
    .map((item) => item.date);
  const dailyDataLen = dates.length;

  const xAxisMax = new Date(
    new Date(dates[dailyDataLen - 1]).setHours(24, 0, 0, 0)
  ).toISOString();

  const xAxisMin = new Date(
    new Date(dates[0]).setHours(-1, 0, 0, 0)
  ).toISOString();

  // Add an extra date one day after the final daily data date
  const chartLabels = [
    new Date(new Date(xAxisMin).setHours(3, 0, 0, 0)).toISOString(),
    ...dates,
    new Date(new Date(xAxisMax).setHours(3, 0, 0, 0)).toISOString(),
  ];

  return {
    xAxisMax,
    xAxisMin,
    chartLabels,
  };
};
