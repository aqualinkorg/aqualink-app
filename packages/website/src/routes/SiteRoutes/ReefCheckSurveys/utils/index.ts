type SegmentsEntity = {
  s1: number;
  s2: number;
  s3: number;
  s4: number;
};

export function segmentsTotalSortComparator<T extends SegmentsEntity>(
  a: T,
  b: T,
) {
  return b.s1 + b.s2 + b.s3 + b.s4 - (a.s1 + a.s2 + a.s3 + a.s4);
}
