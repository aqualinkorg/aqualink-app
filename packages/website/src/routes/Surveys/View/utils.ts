import {
  getSofarDataClosestToDate,
  sameDay,
} from "../../../common/Chart/utils";
import {
  DailyData,
  ValueWithTimestamp,
  TimeSeries,
} from "../../../store/Sites/types";

const getSensorValue = (data?: ValueWithTimestamp[], date?: string | null) =>
  date && data?.[0]
    ? getSofarDataClosestToDate(data, new Date(date), 6)?.value
    : undefined;

export const getCardTemperatureValues = (
  dailyData: DailyData[],
  bottomTemperature: TimeSeries | undefined,
  topTemperature: TimeSeries | undefined,
  date: string | null | undefined
) => {
  const surfaceData = date
    ? dailyData.find((item) => sameDay(date, item.date))
    : undefined;

  return {
    satelliteTemperature: surfaceData?.satelliteTemperature,
    spotterBottom: getSensorValue(bottomTemperature?.spotter?.data, date),
    spotterTop: getSensorValue(topTemperature?.spotter?.data, date),
    hoboBottom: getSensorValue(bottomTemperature?.hobo?.data, date),
    hoboSurface: getSensorValue(topTemperature?.hobo?.data, date),
  };
};
