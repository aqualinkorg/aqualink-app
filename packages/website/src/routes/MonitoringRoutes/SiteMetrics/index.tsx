import React from 'react';
import { Grid, TextField } from '@mui/material';
import makeStyles from '@mui/styles/makeStyles';
import { DatePicker, LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
import { DateTime } from 'luxon';
import monitoringServices, {
  GetMonitoringMetricsResponse,
  MonitoringData,
} from 'services/monitoringServices';
import ChartWithTooltip, {
  ChartWithTooltipProps,
} from 'common/Chart/ChartWithTooltip';
import { Dataset } from 'common/Chart';
import { ArrayElement } from 'utils/types';
import { ValueWithTimestamp } from 'store/Sites/types';
import OneColorSwitch from 'common/OneColorSwitch';
import MonitoringPageWrapper from '../MonitoringPageWrapper';

function transformToDatasets(
  siteInfo: ArrayElement<GetMonitoringMetricsResponse>,
): Dataset[] {
  const monitoringDataKeys: (keyof MonitoringData)[] = [
    'totalRequests',
    'registeredUserRequests',
    'siteAdminRequests',
    'timeSeriesRequests',
    'CSVDownloadRequests',
  ];

  const [
    totalRequests,
    registeredUserRequests,
    siteAdminRequests,
    timeSeriesRequests,
    CSVDownloadRequests,
  ]: ValueWithTimestamp[][] = monitoringDataKeys.map((key) =>
    siteInfo.data.map((x) => {
      return {
        value: Number(x[key]),
        timestamp: x.date,
      };
    }),
  );

  const totalRequestsDataset: Dataset = {
    label: 'total requests',
    data: totalRequests,
    type: 'line',
    unit: '',
    curveColor: '#168dbd',
    displayData: true,
    considerForXAxisLimits: true,
  };

  const registeredUserRequestsDataset: Dataset = {
    label: 'registered users requests',
    data: registeredUserRequests,
    type: 'line',
    unit: '',
    curveColor: 'red',
    displayData: true,
    considerForXAxisLimits: true,
  };

  const siteAdminRequestsDataset: Dataset = {
    label: 'site admin requests',
    data: siteAdminRequests,
    type: 'line',
    unit: '',
    curveColor: 'green',
    displayData: true,
    considerForXAxisLimits: true,
  };

  const timeSeriesRequestsDataset: Dataset = {
    label: 'dashboard visits',
    data: timeSeriesRequests,
    type: 'line',
    unit: '',
    curveColor: 'blue',
    displayData: true,
    considerForXAxisLimits: true,
  };

  const CSVDownloadRequestsDataset: Dataset = {
    label: 'CSV downloads requests',
    data: CSVDownloadRequests,
    type: 'line',
    unit: '',
    curveColor: 'black',
    displayData: true,
    considerForXAxisLimits: true,
  };

  return [
    totalRequestsDataset,
    registeredUserRequestsDataset,
    siteAdminRequestsDataset,
    timeSeriesRequestsDataset,
    CSVDownloadRequestsDataset,
  ];
}

const ONE_DAY = 24 * 60 * 60 * 1000; // One day in milliseconds

function getChartPeriod(siteInfo: ArrayElement<GetMonitoringMetricsResponse>) {
  const { data } = siteInfo;
  const maxDate = data.reduce(
    (max, curr) => (max > curr.date ? max : curr.date),
    new Date(0).toISOString(),
  );
  const minDate = data.reduce(
    (min, curr) => (min < curr.date ? min : curr.date),
    new Date().toISOString(),
  );

  if (minDate === undefined || maxDate === undefined) {
    return undefined;
  }

  const milliseconds =
    new Date(maxDate).getTime() - new Date(minDate).getTime();

  const dataDuration = milliseconds / ONE_DAY;

  if (dataDuration <= 30) {
    return 'day';
  }
  return undefined;
}

function SiteMetrics() {
  const classes = useStyles();

  const [siteId, setSiteId] = React.useState<string>('');
  const [spotterId, setSpotterId] = React.useState<string>('');
  const [monthly, setMonthly] = React.useState<boolean>(false);
  const [startDate, setStartDate] = React.useState<Date>(
    DateTime.now().minus({ months: 3 }).toJSDate(),
  );
  const [endDate, setEndDate] = React.useState<Date>(DateTime.now().toJSDate());

  React.useEffect(() => {
    setSpotterId('');
  }, [siteId]);

  React.useEffect(() => {
    setSiteId('');
  }, [spotterId]);

  const getResult = React.useCallback(
    async (token: string) =>
      (
        await monitoringServices.getMonitoringStats({
          token,
          ...(siteId && { siteIds: [siteId] }),
          ...(spotterId && { spotterId }),
          monthly,
          start: startDate.toISOString(),
          end: endDate.toISOString(),
        })
      ).data,
    [endDate, monthly, siteId, spotterId, startDate],
  );

  const filters = (
    <div className={classes.filtersWrapper}>
      <TextField
        className={classes.filterItem}
        variant="outlined"
        label="Site ID"
        value={siteId}
        onChange={(e) => setSiteId(e.target.value)}
      />
      <TextField
        className={classes.filterItem}
        variant="outlined"
        label="Spotter ID"
        value={spotterId}
        onChange={(e) => setSpotterId(e.target.value)}
      />
      <LocalizationProvider dateAdapter={AdapterDateFns}>
        <DatePicker
          className={classes.filterItem}
          format="MM/dd/yyyy"
          closeOnSelect
          // size="small"
          // showTodayButton
          value={startDate}
          onChange={(e) => e && setStartDate(e)}
          label="start date"
          slotProps={{
            toolbar: {
              hidden: true,
            },
            textField: {
              variant: 'outlined',
            },
          }}
        />
      </LocalizationProvider>

      <LocalizationProvider dateAdapter={AdapterDateFns}>
        <DatePicker
          className={classes.filterItem}
          format="MM/dd/yyyy"
          closeOnSelect
          // size="small"
          // showTodayButton
          value={endDate}
          onChange={(e) => e && setEndDate(e)}
          label="end date"
          slotProps={{
            toolbar: {
              hidden: true,
            },
            textField: {
              variant: 'outlined',
            },
          }}
        />
      </LocalizationProvider>

      <Grid
        container
        alignItems="center"
        spacing={1}
        className={classes.switchContainer}
      >
        <Grid item>weekly</Grid>
        <Grid item>
          <OneColorSwitch
            checked={monthly}
            onChange={(e: any) => {
              setMonthly(e.target.checked);
            }}
            name="checkedC"
          />
        </Grid>
        <Grid item>monthly</Grid>
      </Grid>
    </div>
  );

  return (
    <MonitoringPageWrapper<
      GetMonitoringMetricsResponse,
      React.PropsWithChildren<ChartWithTooltipProps>
    >
      pageTitle="Site Metrics"
      pageDescription="Use only one of Site ID or Spotter ID!"
      ResultsComponent={ChartWithTooltip}
      resultsComponentProps={(res) => ({
        className: classes.chart,
        siteId: Number(siteId),
        surveys: [],
        datasets: transformToDatasets(res[0]),
        temperatureThreshold: null,
        maxMonthlyMean: null,
        background: true,
        hideYAxisUnits: true,
        chartPeriod: getChartPeriod(res[0]),
      })}
      getResult={getResult}
      filters={filters}
      fetchOnEnter
      csvDownload={(token) =>
        monitoringServices.getMonitoringStatsCSV({
          token,
          ...(siteId && { siteIds: [siteId] }),
          ...(spotterId && { spotterId }),
          monthly,
          start: startDate?.toISOString(),
          end: endDate?.toISOString(),
        })
      }
    />
  );
}

const useStyles = makeStyles(() => ({
  chart: {
    width: '100%',
    height: '16rem',
    marginBottom: '3rem',
    marginTop: '1rem',
  },
  filtersWrapper: {
    display: 'flex',
    flexWrap: 'wrap',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    gap: '2rem',
    padding: '2rem',
    flexBasis: '5rem',
  },
  switchContainer: {
    width: 'fit-content',
    height: '3rem',
  },
  filterItem: {
    height: '3rem',
  },
}));

export default SiteMetrics;
