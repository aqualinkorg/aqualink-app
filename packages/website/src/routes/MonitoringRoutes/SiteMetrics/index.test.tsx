import React from 'react';
import configureStore from 'redux-mock-store';
import { mockUser } from 'mocks/mockUser';
import { SnackbarProvider } from 'notistack';
import { mockCollection } from 'mocks/mockCollection';
import { renderWithProviders } from 'utils/test-utils';
import SiteMetrics from '.';

const mockStore = configureStore([]);

describe('Monitoring SiteMetrics Page', () => {
  beforeAll(() => {
    vi.setSystemTime(new Date('2023-11-23T12:00:00'));
  });

  afterAll(() => {
    vi.useRealTimers();
  });

  let element: HTMLElement;

  beforeEach(() => {
    const store = mockStore({
      user: {
        userInfo: mockUser,
        loading: false,
        error: null,
      },
      collection: {
        details: mockCollection,
        loading: false,
        error: null,
      },
    });

    store.dispatch = vi.fn();

    element = renderWithProviders(
      <SnackbarProvider>
        <SiteMetrics />
      </SnackbarProvider>,
      { store },
    ).container;
  });

  it('should render with given state from Redux store', () => {
    expect(element).toMatchSnapshot();
  });
});
