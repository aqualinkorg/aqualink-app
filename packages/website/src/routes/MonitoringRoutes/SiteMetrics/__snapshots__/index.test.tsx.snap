// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monitoring SiteMetrics Page should render with given state from Redux store 1`] = `
<div>
  <div>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 0; visibility: hidden; z-index: 42;"
    >
      <mock-circularprogress
        color="inherit"
      />
    </div>
    <a
      href="/monitoring"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-5 MuiButton-outlinedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          Go back
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </a>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-5 MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Refresh
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <mock-typography
      classname="makeStyles-pageTitle-7"
      variant="h3"
    >
      Site Metrics
    </mock-typography>
    <mock-typography
      classname="makeStyles-pageDescription-8"
      variant="body2"
    >
      Use only one of Site ID or Spotter ID!
    </mock-typography>
    <div
      class="makeStyles-filtersWrapper-2"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-filterItem-4"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Site ID
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Site ID
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-filterItem-4"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Spotter ID
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Spotter ID
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <mock-muipickersutilsprovider>
        <mock-keyboarddatepicker
          autook="true"
          classname="makeStyles-filterItem-4"
          disabletoolbar="true"
          format="MM/dd/yyyy"
          inputvariant="outlined"
          label="start date"
          showtodaybutton="true"
          size="small"
          value="Thu Jul 20 2023 12:00:00 GMT+0000 (Coordinated Universal Time)"
        />
      </mock-muipickersutilsprovider>
      <mock-muipickersutilsprovider>
        <mock-keyboarddatepicker
          autook="true"
          classname="makeStyles-filterItem-4"
          disabletoolbar="true"
          format="MM/dd/yyyy"
          inputvariant="outlined"
          label="end date"
          showtodaybutton="true"
          size="small"
          value="Fri Oct 20 2023 12:00:00 GMT+0000 (Coordinated Universal Time)"
        />
      </mock-muipickersutilsprovider>
      <div
        class="MuiGrid-root makeStyles-switchContainer-3 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          weekly
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <mock-switch
            checked="false"
            classes="[object Object]"
            name="checkedC"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          monthly
        </div>
      </div>
    </div>
    <div
      class="makeStyles-resultsContainer-6"
    />
  </div>
</div>
`;
